### PR TITLE
Improve nan handling in VMC loop

### PR DESCRIPTION
**Pull request by ggoldsh**
Settings for handling nans were quite complex before, but they didn't support the most useful case: namely to stop a run automatically when nans are encountered.

After this update the only options will be
1) Don't check for nans at all
2) Check for nans, and when they are encountered, save a checkpoint and abort the training loop. Also abort the evaluation run if nans are detected during the training.

### CODE REVIEW

**Comment by JiahaoYao** [comment]
tests are failing?

**Comment by ggoldsh** [comment]
Yeah. Partly because of the chex issue but also it looks like my previous version of black was wrong so I got the correct version and ran it again and it undid the formatting changes I was proposing.

File: tests/units/utils/test_checkpoint.py [hunk 1]
```diff
@@ -128,157 +127,6 @@ def track_and_save_best(
     )
 
 
-def _test_nans_checkpointing(
```
Comment (ggoldsh) [inline tests/units/utils/test_checkpoint.py:131]: Since I simplified the logic of the nans checkpointing, I don't think these tests are useful anymore.

**Comment by ggoldsh** [comment]
@JiahaoYao this is now ready for review!

**Pull request merged**